### ISSUE TITLE
Persistent storage on Windows containers

### ISSUE THREAD

**Issue by AndreasLudviksen** [Azure/service-fabric-issues#310]
I use service fabric to deploy a number of services hosted in containers across a cluster.
Some of the services need persistent storage, e.g. Azure File Storage.

As far as I can tell there are no Volume drivers available for Windows.

My other option is to map the external volume on the container hosts in my cluster, and then map that to the container. However, acessing the mapped directory in the container produce an error:

      PS C:\> ls .\certificates\
      ls : The parameter is incorrect.
      At line:1 char:1
      + ls .\certificates\
      + ~~~~~~~~~~~~~~~~~~
          + CategoryInfo          : ReadError: (C:\certificates\:String) [Get-ChildItem], IOException
          + FullyQualifiedErrorId : DirIOError,Microsoft.PowerShell.Commands.GetChildItemCommand

I have also tried to map the external storage during container startup without luck.

Is there a way to create shared,persistent storage between containers in docker for windows?

**Comment by RajeetN** [comment]
How are you mapping the host folder into the container?

**Comment by AndreasLudviksen** [comment]
On the host I use 'net use' to map the folder. 
The mapping to the container is done using the Application manifest 'Volume' directive.

After some investigation, I figured that the 'net use' is session-bound, and this is probably the problem.

I was able to make the volume permanently available by using 'cmdkey' and in the Dockerfile, and UNC paths in the application. Also, if I run the application under a user with the same username/password as the Azure File Storage, the application can access the storage.

It would be much appreciated though, to have a storage driver available for this.
